### PR TITLE
Add Mark a todo as done functionality

### PR DESCRIPTION
**Pull request by DannyvdSluijs**
Fixes issue #1

### CODE REVIEW

**Comment by prolic** [comment]
build failed, please fix php cs

File: src/Model/Todo/Event/TodoWasMarkedAsDone.php [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    private $newStatus;
+
+    /**
+     *
+     * @param TodoStatus $oldStatus
+     * @param TodoId $todoId
+     * @param TodoStatus $newStatus
+     * @return TodoWasMarkedAsDone
+     */
+    public static function fromStatus(TodoStatus $oldStatus, TodoId $todoId, TodoStatus $newStatus)
```
Comment (prolic) [inline src/Model/Todo/Event/TodoWasMarkedAsDone.php]: make todoId first argument, not second please.

File: src/Model/Todo/Todo.php [hunk 2]
```diff
@@ -57,6 +58,14 @@ public static function post($text, UserId $assigneeId, TodoId $todoId)
         return $self;
     }
 
+    public function markAsDone(TodoStatus $status)
+    {
+        if ($this->status->toString() !== TodoStatus::OPEN) {
```
Comment (bweston92) [inline src/Model/Todo/Todo.php]: `if ($this->status->isOpen() === false)` might be better choice.

File: src/Model/Todo/Todo.php [hunk 3]
```diff
@@ -92,6 +101,10 @@ protected function whenTodoWasPosted(TodoWasPosted $event)
         $this->status = $event->todoStatus();
     }
 
+    protected function whenTodoWasMarkedAsDone(Event\TodoWasMarkedAsDone $event)
```
Comment (prolic) [inline src/Model/Todo/Todo.php]: add phpdoc please

File: src/Projection/Todo/TodoProjector.php [hunk 4]
```diff
@@ -47,4 +48,17 @@ public function onTodoWasPosted(TodoWasPosted $event)
...
     }
+
+    /**
+     * @param TodoWasMarkedAsDone $event
+     */
+    public function onTodoWasMarkedAsDone(TodoWasMarkedAsDone $event)
+    {
+        $this->connection->update(Table::TODO, [
+            'status' => $event->newStatus()->toString()
+        ],
```
Comment (prolic) [inline src/Projection/Todo/TodoProjector.php]: one more indent

File: src/Model/Todo/Command/MarkTodoAsDone.php [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+    /**
+     * @return TodoId
+     */
+    public function todoId()
+    {
+        return TodoId::fromString($this->payload['todo_id']);
+    }
+
+
```
Comment (prolic) [inline src/Model/Todo/Command/MarkTodoAsDone.php]: remove these 2 empty lines

File: src/Model/Todo/Event/TodoWasMarkedAsDone.php (hunk 1 above)
Comment (DannyvdSluijs) [inline src/Model/Todo/Event/TodoWasMarkedAsDone.php]: Logical, will update PR when finished.

File: src/Model/Todo/Todo.php (hunk 2 above)
Comment (DannyvdSluijs) [inline src/Model/Todo/Todo.php]: Will do, but should be ```if (!$this->status->isOpen())```

Comment (prolic) [inline src/Model/Todo/Todo.php]: :+1:

File: src/Projection/Todo/TodoProjector.php (hunk 4 above)
Comment (prolic) [inline src/Projection/Todo/TodoProjector.php]: should be

    $this->connection->update(Table::TODO, 
        [
            'status' => $event->newStatus()->toString()
        ],
        [
            'id' => $event->todoId()->toString()
        ]
    );

**Pull request merged**